data/transactions/logic/box.go

```diff
@@ -57,9 +57,14 @@ func (cx *EvalContext) availableBox(name string, operation int, createSize uint6
 				return "", false, fmt.Errorf("box size mismatch %d %d", uint64(len(content)), createSize)
 			}
 		} else {
-			// Since it does not exist, this creation will be dirtying
-			dirty = true
-			cx.available.dirtyBytes += createSize
+			// Since it does not exist, there's 2 cases:
+			// 1. box_create issues create.  `!dirty` should always be true.
+			// 2. box_put issues create.  `!dirty` should never be true.  Otherwise, double counting dirtyBytes.
```

Comment:
I am not following.  If the box did not exist, it can't be dirty.

Let's never speak of the value of `!dirty`, only `dirty`, my brain is unwilling to do double negatives.

So I'm saying, in case 2. When `box_put` happens and the box did not exist, it must be a dirtying write.

Reply:
@jannotti Let me know if the explanation helps clarify.  If not, let's jump on a call.

For case 2, the dirty accounting happens _earlier_ in `opBoxPut` during the 1st `availableBox(..., boxWrite, ...)` check.  

Of course, we may decide that the sequencing ought to be changed.  That's where my _refactoring_ remark comes into play.